### PR TITLE
feat: Add an authentication contract and logging

### PR DESCRIPTION
**Pull request by howjmay**
Each DB Table would hosts a contract `auditLog.sol`. All the DB
operations would be saved in the event log in this `auditLog.sol`.

`authRole.sol` is a contract for authenticating the permission of all
the dropkit service requester. At the same time, `authRole.sol` is the
only one (both contract account and EOA) which can interact with the
contract maintained by each Table.

### CODE REVIEW

File: contract/contracts/auditLog.sol [hunk 1]
```diff
@@ -1,39 +1,51 @@
...
 
-/**
- * @title Storage
- * @dev Store & retrieve value in a variable
- */
-contract auditLog {
-    // The table for mapping DB table name and address which save the the audit log of DB
-    // operations.
-    mapping(string => address) auditLogMap;
+contract SK2AddrTable {
```
Comment (jserv) [inline contract/contracts/auditLog.sol:4]: Why `SK2AddrTable`?

**Review by jserv — CHANGES_REQUESTED** [review]
Explain SK and even document it.

File: contract/contracts/auditLog.sol (hunk 1 above)
Comment (howjmay) [reply contract/contracts/auditLog.sol:4]: SK means secret key. I have mentioned it in the `authRole.sol`

File: contract/contracts/authRole.sol [hunk 2]
```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.5.x;
+
+import "./auditLog.sol";
```
Comment (jserv) [inline contract/contracts/authRole.sol:4]: prefix `./` is not necessary.

File: contract/test/TestAuditLog.sol [hunk 3]
```diff
@@ -2,32 +2,103 @@ pragma solidity 0.5.x;
...
 import "truffle/DeployedAddresses.sol";
-import "../contracts/auditLog.sol";
+import "../contracts/authRole.sol";
 
 contract TestAuditLog {
+    string skOwnerTest = "e0302f799ee2e6bcff366cf395dda8225e0a3ae9250740aeabf8e174a8d55c03";
+    string skMaintainerTest = "aaf4f5s4df5sd47h2ns4i54yk24a65sg7jt7iagj8lo4lj55wfwefjjjukaewqq4";
     string tableNameTest = "TableName";
     address addrTest = 0xc0ffee254729296a45a3885639AC7E10F9d54979;
+    enum ROLE {NULL, VIEWER, MAINTAINER, OWNER}
```
Comment (jserv) [inline contract/test/TestAuditLog.sol:12]: The role might be a bit misleading. How can we distinguish from maintainer and owner?

File: contract/contracts/authRole.sol (hunk 2 above)
Comment (howjmay) [reply contract/contracts/authRole.sol:4]: If I remove the `./` then I will meet the following error, `ParserError: Source "auditLog.sol" not found: File import callback not supported
import "auditLog.sol";`

File: contract/test/TestAuditLog.sol (hunk 3 above)
Comment (howjmay) [reply contract/test/TestAuditLog.sol:12]: MAINTAINER can update the table only. However, a OWNER has the highest permission to do anything.

Comment (howjmay) [reply contract/test/TestAuditLog.sol:12]: I have mentioned the permission in the Postman document

Comment (jserv) [reply contract/test/TestAuditLog.sol:12]: At least, append comments which describe the roles.

**Pull request merged**